client/src/main/java/org/spine3/time/OffsetDateTimes.java

```diff
@@ -0,0 +1,372 @@
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.spine3.time;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.Date;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Calendar.DAY_OF_MONTH;
+import static java.util.Calendar.HOUR;
+import static java.util.Calendar.MILLISECOND;
+import static java.util.Calendar.MINUTE;
+import static java.util.Calendar.MONTH;
+import static java.util.Calendar.SECOND;
+import static java.util.Calendar.YEAR;
+import static org.spine3.time.Calendars.at;
+import static org.spine3.time.Calendars.checkArguments;
+import static org.spine3.time.Calendars.toCalendar;
+import static org.spine3.time.Calendars.toLocalDate;
+import static org.spine3.time.Calendars.toLocalTime;
+import static org.spine3.time.Formats.appendSubSecond;
+import static org.spine3.time.Formats.appendZoneOffset;
+import static org.spine3.time.Formats.dateTimeFormat;
+
+/**
+ * Routines for working with {@link OffsetDateTime}.
+ *
+ * @author Alexander Aleksandrov
+ * @author Alexander Yevsyukov
+ */
+@SuppressWarnings("ClassWithTooManyMethods")
+public final class OffsetDateTimes {
+
+    private OffsetDateTimes() {
+        // Prevent instantiation of this utility class.
+    }
+
+    /**
+     * Obtains current date/time at the passed time zone.
+     */
+    public static OffsetDateTime now(ZoneOffset zoneOffset) {
+        checkNotNull(zoneOffset);
+        final Calendar now = at(zoneOffset);
+        final LocalTime localTime = toLocalTime(now);
+        final LocalDate localDate = toLocalDate(now);
+
+        return create(localDate, localTime, zoneOffset);
+    }
+
+    /**
+     * Creates a new {@code OffsetDateTime} instance with the passed values.
+     */
+    public static OffsetDateTime of(LocalDate date, LocalTime time, ZoneOffset offset) {
+        return create(date, time, offset);
+    }
+
+    private static OffsetDateTime create(LocalDate date, LocalTime time, ZoneOffset offset) {
+        return OffsetDateTime.newBuilder()
+                             .setDate(date)
+                             .setTime(time)
+                             .setOffset(offset)
```

Comment:
See my comment above. I don't think that's what we want to set right away, but rather analyze and adjust before setting the `offset` value.